specfile/tags.py

```diff
@@ -0,0 +1,439 @@
+        rest = itertools.dropwhile(lambda x: x.startswith("#"), reversed(lines))
+        return [c[1:] for c in reversed(list(comments))], list(reversed(list(rest)))
+
+
+class Tags(collections.UserList):
+    """
+    Class that represents all tags in a certain %package section.
+
+    Tags can be accessed by index or conveniently by name as attributes:
+    ```
+    # print name of the first tag
+    print(tags[0].name)
+
+    # set value of Url tag
+    tags.url = 'https://example.com'
+
+    # remove Source1 tag
+    del tags.source1
+    ```
+
+    Attributes:
+        data: List of individual tags.
+    """
+
+    def __init__(
+        self, data: Optional[List[Tag]] = None, remainder: Optional[List[str]] = None
+    ) -> None:
+        super().__init__()
+        if data is not None:
+            self.data = data.copy()
+        self._remainder = remainder.copy() if remainder is not None else []
+
+    def __repr__(self) -> str:
+        data = repr(self.data)
+        remainder = repr(self._remainder)
+        return f"Tags({data}, {remainder})"
+
+    def __getattr__(self, name: str) -> Tag:
+        try:
+            return self.data[self.find(name)]
+        except ValueError:
+            raise AttributeError(name)
+
+    def __setattr__(self, name: str, value: str) -> None:
+        if name.capitalize().rstrip("0123456789") not in TAG_NAMES:
+            return super().__setattr__(name, value)
+        try:
+            self.data[self.find(name)].value = value
+        except ValueError:
+            raise AttributeError(name)
+
+    def __delattr__(self, name: str) -> None:
+        if name.capitalize().rstrip("0123456789") not in TAG_NAMES:
+            return super().__delattr__(name)
+        try:
+            del self.data[self.find(name)]
+        except ValueError:
+            raise AttributeError(name)
+
+    def find(self, name: str) -> int:
+        for i, tag in enumerate(self.data):
+            if tag.name.capitalize() == name.capitalize():
+                return i
+        raise ValueError
+
+    @staticmethod
+    def parse(raw_section: Section, parsed_section: Optional[Section] = None) -> "Tags":
+        """
+        Parses a section into tags.
+
+        Args:
+            raw_section: Raw (unprocessed) section.
+            parsed_section: The same section after parsing.
+
+        Returns:
+            Constructed instance of `Tags` class.
+        """
+
+        def regex_pattern(tag):
+            name = re.escape(tag)
+            index = r"\d?" if tag in ["Source", "Patch"] else ""
+            return fr"^(?P<n>{name}{index})(?P<s>\s*:\s*)(?P<v>.+)"
+
+        tag_regexes = [re.compile(regex_pattern(t), re.IGNORECASE) for t in TAG_NAMES]
+        data = []
+        buffer: List[str] = []
+        for line in raw_section:
+            m = next((m for m in (r.match(line) for r in tag_regexes) if m), None)
+            if m:
+                e = next(
```

Comment:
I find this method a bit hard to read/understand at first sight (took me some time to understand what's going on). Maybe some comments could be helpful here, especially due to the use of short names (which is understandable), maybe a quick comment that we need to parse the tag in both the raw and expanded section.

Reply:
Fair enough, I'll add some comments.